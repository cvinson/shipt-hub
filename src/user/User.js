import React from 'react';
import PropTypes from 'prop-types';
import { Card, CardHeader } from 'material-ui/Card';
import plur from 'plur';

const User = ({ avatar_url, followers, login }) => (
  <Card>
    <CardHeader
      title={<a href={`https://github.com/${login}`}>{login}</a>}
      subtitle={`${followers} ${plur('Follower', followers)}`}
      avatar={avatar_url} />
  </Card>
);

User.PropTypes = {
  avatar_url: PropTypes.string.isRequired,
  followers: PropTypes.number.isRequired,
  login: PropTypes.string.isRequired
};

export default User;
