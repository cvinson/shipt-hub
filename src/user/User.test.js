import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import User from './User';

describe('<User />', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(
      <User
        avatar_url="anAvatarUrl"
        followers={255}
        login="aGithubUser" />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('correctly displays singular follower count', () => {
    const wrapper = shallow(
      <User
        avatar_url="anAvatarUrl"
        followers={1}
        login="aGithubUser" />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  })
});
