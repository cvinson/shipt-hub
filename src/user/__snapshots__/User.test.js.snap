// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<User /> correctly displays singular follower count 1`] = `
<Card
  expandable={false}
  expanded={null}
  initiallyExpanded={false}
>
  <CardHeader
    avatar="anAvatarUrl"
    subtitle="1 Follower"
    title="aGithubUser"
  />
</Card>
`;

exports[`<User /> renders without crashing 1`] = `
<Card
  expandable={false}
  expanded={null}
  initiallyExpanded={false}
>
  <CardHeader
    avatar="anAvatarUrl"
    subtitle="255 Followers"
    title="aGithubUser"
  />
</Card>
`;
